From 21a9cefca2af51765347c4c753bdf986e05b2f35 Mon Sep 17 00:00:00 2001
From: Stephan Hartmann <stha09@googlemail.com>
Date: Sat, 28 Nov 2020 20:04:31 +0000
Subject: [PATCH] add explicit template specialization for String::WriteToFlat

String::WriteToFlat with SharedStringAccessGuardIfNeeded needs
explicit template specialization, because it is used in other
translation units.
---
 v8/src/objects/string.cc | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/v8/src/objects/string.cc b/v8/src/objects/string.cc
index 0455ff4..f43a847 100644
--- a/v8/src/objects/string.cc
+++ b/v8/src/objects/string.cc
@@ -1692,5 +1692,9 @@ template EXPORT_TEMPLATE_DEFINE(V8_EXPORT_PRIVATE) void String::WriteToFlat(
 template EXPORT_TEMPLATE_DEFINE(V8_EXPORT_PRIVATE) void String::WriteToFlat(
     String source, uint8_t* sink, int from, int to);
 
+template EXPORT_TEMPLATE_DEFINE(V8_EXPORT_PRIVATE) void String::WriteToFlat(
+    String source, uint16_t* sink, int from, int to,
+    const SharedStringAccessGuardIfNeeded& access_guard);
+
 }  // namespace internal
 }  // namespace v8
-- 
2.26.2

